dissect/target/plugins/os/unix/locale.py

```diff
@@ -0,0 +1,127 @@
+    def language(self):
+        """
+        Get the configured locale(s) of the system.
+        """
+        locale_paths = ["/etc/default/locale", "/etc/locale.conf"]
+
+        for locale_path in locale_paths:
+            try:
+                for line in self.target.fs.path(locale_path).open("rt"):
+                    if "LANG=" in line:
+                        return line.replace("LANG=", "").replace('"', "").strip()
+                return
+            except FileNotFoundError:
+                pass
+
+    @export(record=UnixKeyboardRecord)
+    def keyboard(self):
+        """
+        Get the keyboard layout(s) of the system.
+        """
+
+        # FIXME: Shorten this dirty unix keyboard spaghetti code
+
+        k = {"LAYOUT": None, "MODEL": None, "VARIANT": None, "OPTIONS": None, "BACKSPACE": None}
+
+        # /etc/default/keyboard
+        XKB = ["XKBMODEL", "XKBLAYOUT", "XKBVARIANT", "XKBOPTIONS", "BACKSPACE"]
+        try:
+            for line in self.target.fs.path("/etc/default/keyboard").open("rt"):
+                for x in XKB:
+                    if x in line:
+                        k[x.replace("XKB", "")] = line.replace(x + "=", "").replace('"', "").strip()
+
+            yield UnixKeyboardRecord(
+                layout=k["LAYOUT"],
+                model=k["MODEL"],
+                variant=k["VARIANT"],
+                options=k["OPTIONS"],
+                backspace=k["BACKSPACE"],
+                _target=self.target,
+            )
+
+        except FileNotFoundError:
+            pass
+
+        if k["LAYOUT"] is not None:
+            return
+
+        # /etc/vconsole.conf
+        try:
+            for line in self.target.fs.path("/etc/vconsole.conf").open("rt"):
+                if "KEYMAP" in line:
+                    k["LAYOUT"] = line.replace("KEYMAP=", "").replace('"', "").strip()
+                    yield UnixKeyboardRecord(
+                        layout=k["LAYOUT"],
+                        _target=self.target,
+                    )
+        except FileNotFoundError:
+            pass
+
+        if k["LAYOUT"] is not None:
+            return
+
+        # /etc/X11/xorg.conf.d/00-keyboard.conf
+        try:
+            for file_ in self.target.fs.path("/etc/X11/xorg.conf.d/").glob("*-keyboard.conf"):
+                for line in file_.open("rt"):
+                    for x in XKB:
+                        if x in line.upper():
+                            k[x.replace("XKB", "")] = (
+                                line.lower().replace('"', "").replace("option " + x.lower(), "").strip()
+                            )
+
+                yield UnixKeyboardRecord(
+                    layout=k["LAYOUT"],
+                    model=k["MODEL"],
+                    variant=k["VARIANT"],
+                    options=k["OPTIONS"],
+                    backspace=k["BACKSPACE"],
+                    _target=self.target,
+                )
+        except FileNotFoundError:
+            pass
```

Comment:
I would suggest something like this:

```suggestion
        k = {}
        paths = ["/etc/default/keyboard", "/etc/vconsole.conf"] + list(
            self.target.fs.glob("/etc/X11/xorg.conf.d/*-keyboard.conf")
        )

        for path_ in paths:
            if (path := self.target.fs.path(path_)).exists():
                for line in path.open("rt"):
                    if len(key_value := line.split("=")) == 2:
                        k[key_value[0].replace("XKB", "").replace("KEYMAP", "LAYOUT")] = (
                            key_value[1].replace('"', "").strip()
                        )

                yield UnixKeyboardRecord(
                    layout=k.get("LAYOUT"),
                    model=k.get("MODEL"),
                    variant=k.get("VARIANT"),
                    options=k.get("OPTIONS"),
                    backspace=k.get("BACKSPACE"),
                    _target=self.target,
                )
                # note that you might still want to exit if LAYOUT is not None. Add if appropriate 
```